torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -860,11 +862,17 @@ void AliasDb::makePointerTo(const Value* from, const Value* to) {
 
   // covariant type containers can be point to types which are not
   // also mutable/immutable because we unify the contained types
+  // Any can point to/from immutable types
```

Comment:
I had to read this 3 times to figure out that this is just an early-out if one of the types in immutable.  I also struggled to understand what the purpose of this assert is. It seems to be checking that isMutableTypeInternal agrees with some subtyping rule, but I can't tell how from the code. from_kind is unused.